### ISSUE TITLE
Async loading and reloading

### ISSUE THREAD

**Issue by phaazon** [phaazon/warmy#15]
This will be a bit tricky but with a `Future` interface, it should be okay.

**Comment by edwin0cheng** [comment]
+1 for this feature, I would like to use warmy in our wasm game engine too. 
But it need to be async loading since wasm resources are async in nature.

**Comment by phaazon** [comment]
Yep. I have an idea in mind on how to do it. Basically, I would like to keep both interface (sync and async). The async one is not very different:

- Instead of returning a `Result<Res<T>, _>`, the `Store::get` method would return a `Result<impl Future<Item = ARes<T>>, _>`.
- `ARes` would be the same as a `Res` but with `Arc`.

The missing parts are that I would like to implement an IO-bound thread instead of letting all asynchronous threads deal with the IO stuff. That’s a bit tricky, because all those threads would then be some kind of *memory buffer consumers* instead of *paths or keys consumers*. A bit tricky. :)

**Comment by edwin0cheng** [comment]
That's good to know, however, I would like to know if it is possible to allow an custom executor ? Since wasm do not support threading, which the only option are polling :(

**Comment by phaazon** [comment]
What do you mean custom executor? I plan to add a way to plugin event loops and custom “event sources”, if it’s what you’re referring to. :upside_down_face:
